pyomo/bilevel/plugins/lcp.py

```diff
@@ -287,7 +286,7 @@ def _add_optimality_conditions(self, instance, submodel):
                         exp -= B2_[uid] * lb_dual
                     if not ub_dual is None:
                         exp += B2_[uid] * ub_dual
-            if type(exp) in six.integer_types or type(exp) is float:
```

Comment:
There is absolutely no reason that this is in here (I think it migrated over from work I was doing on `remove-six`), so I'll revert it if you want. But there's also no harm in keeping it.